### PR TITLE
Nova Forward ports (easy)

### PR DESCRIPTION
**Pull request by huitseeker**
Sources:
- https://github.com/microsoft/Nova/pull/303
- https://github.com/microsoft/Nova/pull/308
- https://github.com/microsoft/Nova/pull/309

Overall:
- QoL improvements in naming,
- stricter testing of R1CS shape during Nova's public parameters setup 

> [!NOTE]
> Companion PR at https://github.com/lurk-lab/lurk-rs/pull/1145

### CODE REVIEW

File: src/bellpepper/test_shape_cs.rs [hunk 1]
```diff
@@ -14,6 +14,7 @@ use ff::{Field, PrimeField};
 #[derive(Clone, Copy)]
 struct OrderedVariable(Variable);
 
+#[allow(unused)]
```
Comment (storojs72) [inline src/bellpepper/test_shape_cs.rs:17]: Do we need this one if it unused?

File: src/lib.rs [hunk 2]
```diff
@@ -1115,36 +1120,36 @@ mod tests {
     test_pp_digest_with::<PallasEngine, _, _, EE<_>, EE<_>>(
       &TrivialCircuit::default(),
       &TrivialCircuit::default(),
-      &expect!["582db42439c0dcfc60d24b023ab83d81d97382ac2efa883c6f23147345efeb01"],
+      &expect!["e5a6a85b77f3fb958b69722a5a21bf656fd21a6b5a012708a4b086b6be6d2b03"],
```
Comment (storojs72) [inline src/lib.rs:1123]: Why do we have changed test vectors?

**Review by storojs72 — COMMENTED** [review]
I'm fine with merging this, when CI passes

**Comment by huitseeker** [comment]
@storojs72 As the note in the PR description indicates, the failure in CI is expected, normal, and fixed by the companion PR.

File: src/bellpepper/test_shape_cs.rs (hunk 1 above)
Comment (huitseeker) [reply src/bellpepper/test_shape_cs.rs:17]: Will remove the annotation.

File: src/lib.rs (hunk 2 above)
Comment (huitseeker) [reply src/lib.rs:1123]: The changes to the R1CS coefficient matrix in src/bellpepper/r1cs.rs change the `R1CSWithArity` (ex-`R1CSShape`), and therefore the public parameters, and therefore their hash.

**Review by mpenciak — APPROVED** [review]

**Review by adr1anh — APPROVED** [review]

**Review by storojs72 — APPROVED** [review]

**Pull request merged**